Packs/Base/Scripts/DBotFindSimilarIncidents/dbot_find_similar_incidents_test.py

```diff
@@ -0,0 +1,198 @@
+    mocker.patch.object(demisto, 'executeCommand', side_effect=executeCommand)
+    res = main()
+    assert ('empty_current_incident_field' not in res.columns)
+    assert (res['Identical indicators'] == ["", "", ""]).all()
+    assert check_exist_dataframe_columns('similarity indicators', 'final_score', 'ID', 'created', 'name', df=res)
+    assert (res['similarity indicators'] == [0.0, 0.0, 0.0]).all()
+
+
+def test_main_no_fetched_incidents_found(mocker):
+    global SIMILAR_INDICATORS, FETCHED_INCIDENT
+    FETCHED_INCIDENT = []
+    mocker.patch.object(demisto, 'args',
+                        return_value={
+                            'incidentId': 12345,
+                            'similarTextField': 'incident.commandline, commandline, command, '
+                                                'empty_current_incident_field, empty_fetched_incident_field',
+                            'similarCategoricalField': 'signature, filehash',
+                            'similarJsonField': 'CustomFields',
+                            'limit': 10000,
+                            'fieldExactMatch': '',
+                            'fieldsToDisplay': 'filehash, destinationip, closeNotes, sourceip, alertdescription',
+                            'showSimilarity': True,
+                            'confidence': 0.2,
+                            'maxIncidentsToDisplay': 100,
+                            'query': '',
+                            'aggreagateIncidentsDifferentDate': 'True',
+                            'includeIndicatorsSimilarity': 'True'
+                        })
+    mocker.patch.object(demisto, 'dt', return_value=None)
+    mocker.patch.object(demisto, 'executeCommand', side_effect=executeCommand)
+    res = main()
+    assert (not res)
+
+
+def test_keep_high_level_field():
+    incidents_field = ['xdralerts.comandline', 'commandline', 'CustomsFields.commandline']
+    res = ['xdralerts', 'commandline', 'CustomsFields']
+    assert keep_high_level_field(incidents_field) == res
+
+
+def test_preprocess_incidents_field():
+    assert preprocess_incidents_field('incident.commandline', PREFIXES_TO_REMOVE) == 'commandline'
+    assert preprocess_incidents_field('commandline', PREFIXES_TO_REMOVE) == 'commandline'
+
+
+def test_check_list_of_dict():
+    assert check_list_of_dict([{'test': 'value_test'}, {'test1': 'value_test1'}]) is True
+    assert check_list_of_dict({'test': 'value_test'}) is False
+
+
+def test_recursive_filter():
+    pass
+
+
+def test_match_one_regex():
+    assert match_one_regex('123.123.123.123', [REGEX_IP]) is True
+    assert match_one_regex('123.123.123', [REGEX_IP]) is False
+    assert match_one_regex('abc', [REGEX_IP]) is False
+    assert match_one_regex(1, [REGEX_IP]) is False
+
+
+def test_normalize_command_line():
+    assert normalize_command_line('cmd -k IP=1.1.1.1 [1.1.1.1]') == 'cmd -k ip = IP IP'
+    assert normalize_command_line('powershell "remove_quotes"') == 'powershell remove_quotes'
+
```

Comment:
please add test case when all the incident fields are missing, in this case you should return message with the missing fields